### PR TITLE
Remove unused deposits/overview API endpoint

### LINKED ISSUE #8179: The `deposits/overview` client REST API endpoint is no longer used and should be removed
### Describe the bug

The `/deposits/overview` is no longer used by the WooPayments client (since WooPayments [2.5](https://github.com/Automattic/woocommerce-payments/releases/tag/2.5.0)). We should remove this endpoint from the client to ensure we're not unnecessarily maintaining redundant code/endpoints.

https://github.com/Automattic/woocommerce-payments/blob/59a85492d0372964127a17f5230b2eb2237fb034/includes/admin/class-wc-rest-payments-deposits-controller.php#L47-L55

REST API documentation must also be updated to reflect these changes.

https://github.com/Automattic/woocommerce-payments/blob/8512583463697730cb901ff318e17b9ffb089494/docs/rest-api/source/includes/wp-api-v3/deposits.md#L176-L188

### Additional information

* Related to server issue for removal of this endpoint https://github.com/Automattic/woocommerce-payments-server/issues/4489
* Related to client issue for removal of data-fetching code related to `/deposits/overview/` #8047

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #8179 

#### Changes proposed in this Pull Request

Remove unused `deposits/overview` API endpoint and associated documentation.

This API has been unused by the client since WooPayments 2.5. The associated frontend code was removed in #8178. 
The `deposits/overview` endpoint proxied the request to the server, logs were checked and confirmed that no recent version of WooPayments has been using this API (ie private use)

#### Testing instructions

As this PR is only removing unused API's, there is nothing to test.
Steps taken
 - Searched for usages of removed method including unit tests.
 - Reviewed logs
 - Reviewed code search results for `/deposit.overview/`

-------------------

- [x] Run `npm run changelog`.
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8207` or branch name `fix/8179-remove-deposits-api` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8179-remove-deposits-api) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6f660c54601a3e5c8a046972ad6799c3512f8262
- Build time: 2024-02-18 22:55:11 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/fix-8179-remove-deposits-api [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
+
+Remove unused `deposits/overview` API
```
Comment (haszari) [inline changelog/fix-8179-remove-deposits-api]: This is not super clear, though maybe we don't need a changelog.

The change here is that we are removing this unused API and dead code related to it in various layers – JS/front end, plugin client, and server. Changelogs for each part of that are confusing IMO!

**Review by haszari — APPROVED** [review]
Looks good!

I tested payments overview, deposits list, deposits export, and deposit detail. Didn't see any unintended side-effects of removing this dead code 👌🏻  

Clarifying the sequence here, since it confused me initially. 

- We are removing `GET /deposits/overview` API as it is no longer used.
- To do that, we need 3 PRs across 2 repos:
  - WooPayments plugin
    - Remove front-end JS for calling the API #8178
    - Remove store backend (plugin/client) wrapper for the server API, and API docs (this PR)
  - WooPayments server
    - Remove the actual API endpoint and the code that powers it https://github.com/Automattic/woocommerce-payments-server/issues/4489

And, it's important we do those pieces in order - server must be last. I see that is acknowledged implicitly in some of these PRs e.g. "related issue for removing X".

Couple of observations:

- We could use a server epic to coordinate this work (the first two PRs are steps toward removing server API). Then we can be explicit about the dependencies & order, so reviewing each chunk is straightforward. 
- There's a lot of boilerplate for each API! To be expected, just acknowledging this architecture has a cost.

**Comment by brucealdridge** [comment]
> * Remove the actual API endpoint and the code that powers it https://github.com/Automattic/woocommerce-payments/issues/4489

This is the incorrect issue. The issue you have linked to is the client repo.

File: changelog/fix-8179-remove-deposits-api (hunk 1 above)
Comment (brucealdridge) [reply changelog/fix-8179-remove-deposits-api]: I think we do need a changelog. 

#8178 was removing dead front-end code that isn't exposed anywhere. This PR removes an API that 3rd parties could be using.

What is unclear about it? Is it the wording?

Comment (haszari) [reply changelog/fix-8179-remove-deposits-api]: > What is unclear about it? Is it the wording?

Just could be more specific – it assumes the reader understands the whole thing that is getting removed (of which this is a part).

> This PR removes an API that 3rd parties could be using.

Write with those potential clients in mind – what do they depend on that is changing? Be specific about the kind of API – is this an HTTP / REST API exposed on the merchant site? Or a mirror of a WooPayments server API? (Or a hook/WP API – api can have many meanings?) If it's potentially used, we should be clearer (and probably not focus on the fact that _we_ aren't using the API).

**Comment by haszari** [comment]
> This is the incorrect issue. The issue you have linked to is the client repo.

Doh! Now fixed. I was trying to confirm the sequence here since all these PRs/issues are solving a single problem and we need to couple them somewhat.

**Pull request merged**